From 3a3050a687b80f8150caabec885f72e26f743947 Mon Sep 17 00:00:00 2001
From: Alan Newberger <alann@google.com>
Date: Fri, 6 May 2016 14:31:02 -0700
Subject: [PATCH 107/173] Grant read URI permission for playback of video
 capture

Current implementation of video capture UI configured MediaRecorder to
output directly into supplied URI, which per recommendation should be
a content: URI. We pass this URI to another app to play back the video,
and should pass along the read permission.

Bug: 19675889

Change-Id: Idfc0ee42422dde5d921e81f098d4697e652ac689
---
 src/com/android/camera/VideoModule.java | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/com/android/camera/VideoModule.java b/src/com/android/camera/VideoModule.java
index a7efae0e7..1d34990bd 100644
--- a/src/com/android/camera/VideoModule.java
+++ b/src/com/android/camera/VideoModule.java
@@ -786,6 +786,7 @@ public class VideoModule implements CameraModule,
 
     private void startPlayVideoActivity() {
         Intent intent = new Intent(Intent.ACTION_VIEW);
+        intent.setFlags(Intent.FLAG_GRANT_READ_URI_PERMISSION);
         intent.setDataAndType(mCurrentVideoUri, convertOutputFormatToMimeType(mProfile.fileFormat));
         try {
             mActivity
-- 
2.17.1

